plugins/chat/spec/system/transcript_spec.rb

```diff
@@ -0,0 +1,247 @@
+        expect(page).to have_css(".chat-message-actions .more-buttons")
+        find(".chat-message-actions .more-buttons").click
+        find(".select-kit-row[data-value=\"selectMessage\"]").click
+      end
+    end
+  end
+
+  def cdp_allow_clipboard_access!
+    cdp_params = {
+      origin: page.server_url,
+      permission: {
+        name: "clipboard-read",
+      },
+      setting: "granted",
+    }
+    page.driver.browser.execute_cdp("Browser.setPermission", **cdp_params)
+
+    cdp_params = {
+      origin: page.server_url,
+      permission: {
+        name: "clipboard-write",
+      },
+      setting: "granted",
+    }
+    page.driver.browser.execute_cdp("Browser.setPermission", **cdp_params)
+  end
+
+  def read_clipboard
+    page.evaluate_async_script("navigator.clipboard.readText().then(arguments[0])")
+  end
+
+  def click_selection_button(button)
+    selector =
+      case button
+      when "quote"
+        "#chat-quote-btn"
+      when "copy"
+        "#chat-copy-btn"
+      when "cancel"
+        "#chat-cancel-selection-btn"
+      when "move"
+        "#chat-move-to-channel-btn"
+      end
+    within(".chat-selection-management-buttons") { find(selector).click }
+  end
+
+  def copy_messages_to_clipboard(messages)
+    messages = Array.wrap(messages)
+    messages.each { |message| select_message(message) }
+    expect(chat_channel_page).to have_selection_management
+    click_selection_button("copy")
+    expect(page).to have_content("Chat quote copied to clipboard")
+    clip_text = read_clipboard
+    expect(clip_text.chomp).to eq(generate_transcript(messages, current_user))
+    clip_text
+  end
+
+  def generate_transcript(messages, acting_user)
+    messages = Array.wrap(messages)
+    ChatTranscriptService
+      .new(messages.first.chat_channel, acting_user, messages_or_ids: messages.map(&:id))
+      .generate_markdown
+      .chomp
+  end
+
+  describe "copying quote transcripts with the clipboard" do
+    before { cdp_allow_clipboard_access! }
+
+    it "quotes a single chat message into a topic " do
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
+
+      clip_text = copy_messages_to_clipboard(chat_message_5)
+
+      # post transcript in topic
+      topic_page.visit_topic_and_open_composer(topic)
+      topic_page.fill_in_composer("This is a new post!\n\n" + clip_text)
+      within ".d-editor-preview" do
+        expect(page).to have_css(".chat-transcript")
+      end
+      topic_page.send_reply
+      expect(page).to have_content("This is a new post!")
+      within topic_page.post_by_number(topic.posts.reload.last.post_number) do
+        expect(page).to have_css(".chat-transcript")
+      end
+    end
+
+    it "quotes multiple chat messages into a topic" do
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
```

Comment:
I just use `page` in these cases, doing `chat_channel_page` doesn't bring anything AFAIK